demisto_sdk/commands/ansible_codegen/ansible_codegen.py

```diff
@@ -0,0 +1,674 @@
+                    continue
+                value = str(value).replace("\n", "\"")
+                value = str(value).replace("\\", "\\\\")
+                example_command += "{}=\"{}\" ".format(arg, value)
+
+        self.example_commands.append(example_command)
+        return self.example_commands
+
+    def save_command_examples(self, directory: str) -> str:
+        """
+        Writes command examples to a file.
+        Args:
+            directory: The directory to save the file to.
+
+        Returns:
+            command_examples_file: The path to the command_examples file.
+        """
+        self.print_with_verbose('Creating command_examples file...')
+        command_examples_file = os.path.join(directory, 'command_examples')
+        try:
+            with open(command_examples_file, 'w') as fp:
+
+                for module in self.ansible_modules:
+                    self.generate_command_example(module=module)
+
+                fp.write('\n'.join(self.example_commands))
+                return command_examples_file
+        except Exception as err:
+            print_error(f'Error writing {command_examples_file} - {err}')
+            raise
+
+    def save_package(self) -> tuple:
+        """
+        Creates a package for the integration including python, yaml, image and description files.
+
+        Returns:
+            code_path: The path to the python code file.
+            yml_path: the path to the yaml file.
+            image_path: The path to the image file.
+            desc_path: The path to the description file.
+            examples_path: The path to the examples file.
+        """
+        code_path = self.save_python_code(self.base_path)
+        yml_path = self.save_yaml(self.base_path)
+        image_path, desc_path = self.save_image_and_desc(self.base_path)
+        examples_path = self.save_command_examples(self.base_path)
+        return code_path, yml_path, image_path, desc_path, examples_path
+
+    def print_with_verbose(self, text: str):
+        """
+        Prints a text verbose is set to true.
+        Args:
+            text: The text to print.
+        """
+        if self.verbose:
+            print(text)
+
+    def get_command_name(self, module: str) -> str:
+        """
+        Determines what the XSOAR command name should be fore the module.
+
+        Returns:
+            command_name: XSOAR friendly command name
+        """
+        # In case ansible module has been provided in collection namespace form. We want just the module name
+        module = module.split(".")[-1]
+
+        command_name = ""
+        # Don't double up with the prefix, if the module name already has the prefix
+        if to_kebab_case(module).startswith(f"{self.command_prefix}-"):
+            command_name = to_kebab_case(module)
+        else:
+            command_name = f"{self.command_prefix}-{to_kebab_case(module)}"
+        return command_name
+
+    def get_yaml_commands(self) -> list:

```

Comment:
I think the function is too long and complex. Let's split it wherever it makes sense, extract code to helper functions, and keep things on one level of abstraction.
(For example, you can make a helper function for the `remove ansible link markup` section or for handling the args/outputs and so on...)